### PR TITLE
Docs: Prometheus metrics browser

### PR DESCRIPTION
**Pull request by davkal**
Describing the metrics browser and what it's for. Added limitations at the end.

### CODE REVIEW

File: docs/sources/datasources/prometheus.md [hunk 1]
```diff
@@ -76,6 +76,26 @@ To show a horizontal line across the whole graph, add a series override and sele
...
 | `Exemplars`        | Run and show exemplars in the graph.                                                                                                                                                                                                                                                                                                                                                                                                       |
 
+### Metrics browser
+
+The metrics browser allows you to quickly find metrics and select relevant labels to build basic queries.
+When you open the browser you will see all available metrics and labels.
+If supported by your Prometheus, each metric will show its HELP and TYPE as a tooltip.
+When you select a metric, the browser will narrow down the available labels to show only the ones applicable to the metric.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/prometheus.md:84]:
```suggestion
When you select a metric, the browser narrows down the available labels to show only the ones applicable to the metric.
```

File: docs/sources/datasources/prometheus.md [hunk 2]
```diff
@@ -76,6 +76,26 @@ To show a horizontal line across the whole graph, add a series override and sele
...
+You can then select one or more labels for which the available label values will be shown in lists in the bottom section.
+You can select one or more values for each label to tighten your query scope.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/prometheus.md:86]:
```suggestion
You can then select one or more labels for which the available label values are shown in lists in the bottom section.
Select one or more values for each label to tighten your query scope.
```

File: docs/sources/datasources/prometheus.md [hunk 3]
```diff
@@ -76,6 +76,26 @@ To show a horizontal line across the whole graph, add a series override and sele
...
+
+Pro tip: If you do not remember a metric name to start with, you can also select a few labels first and find relevant label values. Once you select a few, the set of applicable metrics is reduced, making it easy to find the metric you were looking for.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/prometheus.md:88]:
```suggestion
>**Note:** If you do not remember a metric name to start with, you can also select a few labels first, to narrow down the list and then find relevant label values. 
```

File: docs/sources/datasources/prometheus.md [hunk 4]
```diff
@@ -76,6 +76,26 @@ To show a horizontal line across the whole graph, add a series override and sele
...
+
+All lists in the metrics browser have a search field above them to quickly filter for metrics or labels that match a certain string. Notice that the values section only has one search field. Its filtering applies to all labels to help you find values across labels once they have been selected, e.g., among your labels `app`, `job`, `job_name` only one might contain the value you are looking for.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/prometheus.md:90]:
```suggestion
All lists in the metrics browser have a search field above them to quickly filter for metrics or labels that match a certain string. The values section only has one search field. Its filtering applies to all labels to help you find values across labels once they have been selected, for example, among your labels `app`, `job`, `job_name` only one might with the value you are looking for.
```

File: docs/sources/datasources/prometheus.md [hunk 5]
```diff
@@ -76,6 +76,26 @@ To show a horizontal line across the whole graph, add a series override and sele
...
+
+Once you are happy with your query, you can click on "Use query" to run the query. The button "Use as rate query" will add a `rate(...)[$__interval]` around your query to help write queries for counter metrics.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/prometheus.md:92]:
```suggestion
Once you are satisfied with your query, click "Use query" to run the query. The button "Use as rate query" adds a `rate(...)[$__interval]` around your query to help write queries for counter metrics.
```

File: docs/sources/datasources/prometheus.md [hunk 6]
```diff
@@ -76,6 +76,26 @@ To show a horizontal line across the whole graph, add a series override and sele
...
+The "Validate selector" button will check with Prometheus how many time series are available for that selector.
+
+Limitations: The metrics browser has a hard limit of 10,000 labels (keys) and 50,000 label values (including metric names). If Prometheus returns more results the browser will continue functioning, but the result sets will be cut off above those limits.
```
Comment (achatterjee-grafana) [inline docs/sources/datasources/prometheus.md:95]:
```suggestion
#### Limitations 

The metrics browser has a hard limit of 10,000 labels (keys) and 50,000 label values (including metric names). If your Prometheus instance returns more results, the browser will continue functioning. However, the result sets will be cut off above those maximum limits.
```

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added doc change requests.

**Comment by davkal** [comment]
Thanks for the remarks, what's the next step?

**Review by achatterjee-grafana — APPROVED** [review]
LGTM! We do not need to backport the changes to 8.0 GA, Torkel will backport all doc changes on Monday. Post merge, you can see the new changes in: https://grafana.com/docs/grafana/next/

**Comment by oddlittlebird** [comment]
Please add this to What's new in Grafana v8.0 and add links.

**Comment by achatterjee-grafana** [comment]
@davkal, I will add the content to What's New, FYI.

**Pull request merged**